slack/tool.gpt

```diff
@@ -187,6 +187,13 @@ When you use the search_messages tool, you can use normal Slack search filters.
 
 ## End of instructions for using Slack tools
 
+---
+Name: Slack User Context
+Type: context
+Credential: ./credential
+
+#!/usr/bin/env node ${GPTSCRIPT_TOOL_DIR}/index.js userContext
```

Comment:
Will this get called for every tool call now? Seems like it only needs to be called once per thread in most cases.

Reply:
I'm not sure. I tried to set this up the same way that the existing `Slack Context` tool works. The only real difference between the two is that this one runs code (and uses a credential), whereas the other one is just sys.echo.

Reply:
Pretty sure it will be called every time - although perhaps cache will catch it?

Reply:
Is there a way I should structure this differently to avoid the excess calls?

Reply:
I had to do something similar for hubspot in a few places.. instead of making the context be a code tool like this, I just added context tool that told it to call a specific tool to get info whenever it was doing operations that needed it. So for slack, you might have a context tool that says `Call the "Get User Context" tool to retrieve information about the currently signed in user including timezone, blah, blah. You must call this tool whenever a request is made that relates to dates/times blah blah blah...`

Reply:
Of course it also looks like I did this exact same thing to look up user identity in the hubspot tools, so really YMMV if it causes any performance issues.

As an example though - https://github.com/otto8-ai/tools/blob/8f1d8906a3688210c49d1b63cee520a44581cd83/apis/hubspot/crm/write/tool.gpt#L18

Reply:
For the `Outlook Calendar` bundle I shared the `Get Default Timezone` tool from the context and added [this line](https://github.com/otto8-ai/tools/blob/c03ab65f0d22660e733a897db4486582ddbc5974/outlook/calendar/tool.gpt#L140) to coax the LLM to get the default timezone only when necessary.